frontend/src/components/Form.js

```diff
@@ -41,11 +41,45 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
+/**
+ * Validates from inputs on submission and alerts user on error
+ * @param {String} role
+ * @param {String} productArea
+ * @param {Number} duration
+ * @param {Number} timeAvailableUntilMilliseconds
+ * @param {Number} currentTimeMilliseconds
+ * @return {Boolean} true or false based on validity of inputs
+ */
+export function validateFormInputs(
+    role,
+    productArea,
+    duration,
+    timeAvailableUntilMilliseconds,
+    currentTimeMilliseconds,
+) {
+  const durationMilliseconds = duration * 60000;
+
+  if (isNaN(timeAvailableUntilMilliseconds)) {
+    alert('Please select a valid date.');
+    return false;
+  } else if (currentTimeMilliseconds + durationMilliseconds >=
+    timeAvailableUntilMilliseconds) {
+    // Check if a meeting is possible with the provided inputs
+    alert('Please select an larger time availability window');
+    return false;
+  } else if (productArea === '' || role === '') {
+    alert('Please select options for all form fields to submit!');
```

Comment:
For these two fields, once we have "no preference" match checkbox, we should allow them to be empty, right?

Reply:
+ 1

Reply:
I think we were discussing that we want to require these fields regardless so that the database doesn't have empty entries, but I think is a great suggestion. I will follow up with the rest of the pod on thoughts and update. Thanks

Reply:
Also wondering about this. Maybe the default option can be something like "Prefer not to say" to make it clear to the user that declining to pick something will have an impact on matching.

Reply:
After discussing we came to this conclusion: We need to store the Role and PA  in case the other person they would match has a preference. For example, if Person A specifies 'similar' and Person B specifies 'any' and both A and B are indeed similar a match between A and B could occur. But, if B does not specify their Role and PA, then A and B would not be able to match because their is no point of comparison between the two since B did not supply any inputs.

Reply:
Thanks for the explanation Dhruvi. One possible alternative could be to lock the match preference to ANY (along with an explanation) until the user specifies at least one field on which they can be matched. That way a user can still participate even if they don't want to specify role and PA. WDYT?

Reply:
I like this idea! It may cause backend implementation to change, but I think it is a good way to get a user to participate without specifying personal details as you mentioned. I will work on this change and update.

Reply:
Thanks for the discussion. If it needs backend implementation to change, we should use a follow up PR to change frontend and backend together to support empty role/PA field. IMO, a TODO here would be enough for this PR.